newton/sim/builder.py

```diff
@@ -1286,6 +1310,58 @@ def add_joint_d6(
             enabled=enabled,
         )
 
+    def add_equality_constraint(
+        self,
+        constraint_type: Any,
+        body1name: str | None = None,
+        body2name: str | None = None,
+        anchor: Vec3 | None = None,
+        torquescale: float | None = None,
+        relpose: list[float] | None = None,
+        joint1name: str | None = None,
+        joint2name: str | None = None,
```

Comment:
We should be using integers rather than strings here and leave it up to the user to do the look up based on the body or joint keys. Those keys are not guaranteed to be unique anyway.

Reply:
Also, I think it would be better to have separate `add` functions for these constraint types, like `add_constraint_weld()`, etc. so that you only have the input arguments that for those functions that are actually relevant to the constraint, similar to the `add_joint_*()` functions we have.